BlockchainSdk/Common/BaseManager.swift

```diff
@@ -11,19 +11,71 @@ import Combine
 
 @available(iOS 13.0, *)
 class BaseManager: WalletProvider {
-    @Published var wallet: Wallet
+    var wallet: Wallet {
+        get { _wallet.value }
+        set { _wallet.value = newValue }
+    }
+    
     var cardTokens: [Token] = []
     
     var defaultSourceAddress: String { wallet.address }
     var defaultChangeAddress: String { wallet.address }
-    
+
     var cancellable: Cancellable? = nil
-    var walletPublisher: Published<Wallet>.Publisher { $wallet }
+
+    var walletPublisher: AnyPublisher<Wallet, Never> { _wallet.eraseToAnyPublisher() }
+    var statePublisher: AnyPublisher<WalletManagerState, Never> { state.eraseToAnyPublisher() }
+
+    private var latestUpdateTime: Date?
+
+    // TODO: move constant into config
+    private var canUpdate: Bool {
+        if let latestUpdateTime,
+           latestUpdateTime.distance(to: Date()) <= 10 {
+            return false
+        }
+
+        return true
+    }
+
+    private var _wallet: CurrentValueSubject<Wallet, Never>
+    private var state: CurrentValueSubject<WalletManagerState, Never> = .init(.initial)
+    private var loadingPublisher: PassthroughSubject<WalletManagerState, Never> = .init()
 
     init(wallet: Wallet) {
-        self.wallet = wallet
+        self._wallet = .init(wallet)
     }
-    
+
+    func update() {
+        if state.value.isLoading {
+            return
+        }
+
+        guard canUpdate else {
+            didFinishUpdating(error: nil)
+            return
+        }
+
+        state.send(.loading)
+
+        update { [weak self] result in
+            guard let self else { return }
+
+            switch result {
+            case .success:
+                self.didFinishUpdating(error: nil)
+            case .failure(let error):
+                self.didFinishUpdating(error: error)
+            }
+        }
+    }
+
+    func update(completion: @escaping (Result<Void, Error>) -> Void) {}
```

Comment:
```suggestion
    func update(completion: @escaping (Result<Void, Error>) -> Void) {
        fatalError("Must be overrided") 
    }
```

Вообще я думаю надо на POP переезжать потихоньку, и убрать BaseManager, и токены из него, они же не всем нужны)

Reply:
мне тоже POP больше нравится, тут вроде не получалось что-то, поэтому сделал базовый класс